### ISSUE TITLE
Implement content based internal representation

### ISSUE THREAD

**Issue by davids91** [davids91/shocovox#2]
Store a simplified/collapsed view of the enclosed content inside internal nodes. 

- define what to store for a simplified view of the node 
- set a threshold for the percentage/count of cubes need be identical to be able to build a simplified parent representation as not all nodes have a valid simplified view

End result will make it possible to have raytracing on different LOD and is effectively an optimization

**Comment by davids91** [comment]
After #10 , it is possible to extend the simplified representation definition from a single voxel to an array of voxels, essentially creating 3 dimensional mip-mapping

**Comment by davids91** [comment]
Duplicate of #28

**Comment by davids91** [comment]
Upon further reflection, this is not exactly a duplicate, but a needlessly complex solution to a simple problem. Instead dominant voxel, only the albedo shall be present in the simplified representation.
With only color involved, a simple average operation is sufficient to correctly represent data
